repository/repository.go

```diff
@@ -0,0 +1,9 @@
+package repository
+
+import "database/sql"
+
+type repository struct {
+	DB *sql.DB
+}
```

Comment:
you should create a "constructor", e.g.
```go
func NewRepository(db *sql.DB) *repository {
	return &repository{
		db: db,
	};
}
```

Reply:
why?

Reply:
Short answer: because it's a common pattern in Golang to do so.

Long answer:
Let's say you want to do more in the constructor (e.g. adding more [options](https://www.sohamkamani.com/golang/options-pattern/)). Without constructor, it will be painful for the client code to do that. Instead you can create a constructor and put all the options as variadic arguments.

But this one is not super critical though. You can add constructor later if you need. It's just the common pattern in Golang.